File: backend/app/schemas/export.py
from typing import Literal

from pydantic import BaseModel

ExportPreset = Literal[
    "original",
    "youtube",
    "tiktok",
    "instagram_portrait",
    "instagram_square",
]


class ExportRequest(BaseModel):
    media_id: str
    preset: ExportPreset
    output_name: str = "export.mp4"

File: frontend/src/types/export.ts
export type ExportPreset =
  | 'original'
  | 'youtube'
  | 'tiktok'
  | 'instagram_portrait'
  | 'instagram_square'

export interface ExportRequest {
  media_id: string
  preset: ExportPreset
  output_name?: string
}

export interface PresetInfo {
  id: ExportPreset
  label: string
  desc: string
  resolution: string
}
